frontend/src/views/consumer/paymentstatus-modal/index.js

```diff
@@ -70,11 +78,13 @@ export default withTheme(
             path="/consumer/payment"
             children={({ match, location }) => {
               const query = parseQuery(location.search);
+              if (!this.state.eventToLoad) {
+                this.loadEvent(query.event_id);
+              }
               const meaning = match && this.getMeaning(query.status);
               const longText = match && this.getMeaningLong(query.status);
               const foundEvent = match && store.findEvent(query.event_id); // maybe undefined if not found
```

Comment:
`const foundEvent = match && store.findEvent(query.event_id); // maybe undefined if not found
const event = foundEvent && foundEvent.toJSON();`

Isn't this supposed to get the event from store based on eventId?. Then why would we need the additional loadEvent call?. Did you debug it?.

Reply:
Without `loadEvent()` `foundEvent` would be `undefined`. I tested it with @vinujosef